web/src/scheduled_messages.js

```diff
@@ -72,30 +76,30 @@ function hide_scheduled_message_success_compose_banner(scheduled_message_id) {
 }
 
 export function add_scheduled_messages(scheduled_messages) {
-    scheduled_messages_data.push(...scheduled_messages);
+    for (const index in scheduled_messages) {
+        if (scheduled_messages.hasOwnProperty(index)) {
+            scheduled_messages_data["message-" + scheduled_messages[index].scheduled_message_id] =
```

Comment:
Why not just use `scheduled_message_id` as key?

Reply:
I have used "message-" as a prefix for the key because in the sort_scheduled_messages_data() function Object.fromEntries() is used to convert the sorted array of messages to the dictionary and this function internally sorts the keys if the key is a number and then returns the complete object but in case of a string key it does not sort the objects by key.

Reply:
oh! I missed out that we are doing something fundamentally wrong here. We should be using `object` to store sorted values.

* Add sorting to `format` function.
* Remove `sort_scheduled_messages_data()` calls from everywhere. 

That should fix this I guess. So, we can use `scheduled_message_id` as key here now.

Reply:
Yeah that SGTM, let me get back to you with these changes